src/utils.ts

```diff
@@ -266,3 +266,99 @@ export const proxyWithComputed = <T extends object, U extends object>(
   const proxyObject = proxy(initialObject) as T & U
   return proxyObject
 }
+
+export type WatchGet = <T extends object>(value: T) => T
+export type WatchCallback = (get: WatchGet) => (() => void) | void | undefined
+
+let currentCleanups: Set<() => void>
```

Comment:
```suggestion
let currentCleanups: Set<() => void> | undefined
```
This is probably proper typing. Or, we could explicitly initialize it with `null`.

Reply:
The motivation for the lack of "undefined" is to raise TS2454 during compilation. However, it seems that it isn't useful at this point. Will do the change.